### PR TITLE
Hotels: use hotelsAmenities field for filtering

### LINKED ISSUE #1440: Use hotelAmenities instead of deprecated hotelFacilities in all hotels search filter

### PR DESCRIPTION
**Pull request by tbergq**
closes #1440 

hotelFacilities has been deprecated on graphql server

### CODE REVIEW

File: app/hotels/src/allHotels/NewAllHotelsSearch.js [hunk 1]
```diff
@@ -75,8 +75,10 @@ class NewAllHotelsSearch extends React.Component<Props> {
             roomsConfiguration,
           },
           filter: {
-            ...filterParamsRest,
-            hotelFacilities: sanitizeHotelFacilities(hotelAmenities),
+            ...filterParams,
+            hotelAmenities: sanitizeHotelFacilities(
```
[the remark this answers is not in the document]
Comment (tbergq) [reply app/hotels/src/allHotels/NewAllHotelsSearch.js]: Yes, you are right, it is time to rename this function also 👍

**Comment by tbergq** [comment]
@RobinCsl 🚀

File: app/hotels/src/GraphQLSanitizers.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ export const sanitizeDate = (input: ?Date): ?string => {
   return input && DateFormatter(input).formatForMachine();
 };
 
-export const sanitizeHotelFacilities = (hotelFacilities: string[]) => {
+export const sanitizeHotelAmenities = (hotelFacilities: string[]) => {
```
Comment (RobinCsl) [inline app/hotels/src/GraphQLSanitizers.js]: Sorry to be picky, but the variable name should probably also be renamed to erase `Facilities` entirely.

Comment (tbergq) [reply app/hotels/src/GraphQLSanitizers.js]: Picky is good, I will fix it

**Comment by tbergq** [comment]
@RobinCsl 🚀

**Pull request merged**